src/core/xds/grpc/xds_routing.cc

```diff
@@ -150,7 +150,9 @@ bool HeadersMatch(const std::vector<HeaderMatcher>& header_matchers,
 
 bool UnderFraction(const uint32_t fraction_per_million) {
   // Generate a random number in [0, 1000000).
-  const uint32_t random_number = rand() % 1000000;
+  static thread_local absl::BitGen bitgen;
```

Comment:
this doesn't seem to be security sensitive either, let's use a InsecureBitGen here as well.